service/crt/decoder.go

```diff
@@ -0,0 +1,38 @@
+package crt
+
+import (
+	"encoding/json"
+	microerror "github.com/giantswarm/microkit/error"
+
+	"github.com/giantswarm/certificatetpr"
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/watch"
+)
+
+// certificateDecoder is used by the watcher to decode the certificatetpr objects.
+type certificateDecoder struct {
+	decoder *json.Decoder
+	close   func() error
+}
+
+// Decode deserializes a runtime object into a certificatetpr CustomObject.
+func (cd *certificateDecoder) Decode() (watch.EventType, runtime.Object, error) {
+	var e struct {
+		Type   watch.EventType
+		Object certificatetpr.CustomObject
+	}
+
+	if err := cd.decoder.Decode(&e); err != nil {
+		return watch.Error, nil, microerror.MaskAny(err)
+	}
+
+	return e.Type, &e.Object, nil
+}
+
+// Close closes the stream.
+func (cd *certificateDecoder) Close() {
+	err := cd.close()
+	if err != nil {
+		panic(err)
+	}
+}
```

Comment:
Is  there a reason why we provide our own decoder? I see [from another project](https://github.com/zalando-incubator/postgres-operator/blob/d34273543ed9aec2ddf97a611315586c615127bd/pkg/controller/postgresql.go#L71-L78) that you can delegate that to the client library.